dbms/src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -562,6 +578,62 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
 	return {};
 }
 
+ASTPtr InterpreterCreateQuery::createQueryWithoutCluster(ASTCreateQuery & create) const
+{
+	ASTPtr cloned = create.clone();
+	ASTCreateQuery & tmp = typeid_cast<ASTCreateQuery &>(*cloned);
+	tmp.cluster.clear();
+	if (tmp.database.empty())
+		tmp.database = context.getCurrentDatabase();
+	return cloned;
+}
+
+void InterpreterCreateQuery::writeToZookeeper(ASTPtr query, const std::vector<Cluster::Address> & addrs)
+{
+	auto zookeeper = context.getZooKeeper();
+	ASTCreateQuery & create = typeid_cast<ASTCreateQuery &>(*query);
+	std::string table_name = create.database + "." + create.table;
+
+	for (const auto & addr : addrs)
+	{
+		const std::string & path =
+			"/clickhouse/task_queue/ddl/" +
+			addr.host_name +
+			"/create/" + table_name;
```

Comment:
It seems that `task_queue` is not actually a queue.
Queries are reordered. Need to mind that and don't forget to mention in doc.